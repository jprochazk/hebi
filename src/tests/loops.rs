check! {
  print_0_to_9,
  r#"
    i := 0
    loop:
      if i >= 10:
        break
      print i
      i += 1
  "#
}
check! {
  while_print_0_to_9,
  r#"
    i := 0
    while i < 10:
      print i
      i += 1
  "#
}
check! {
  for_print_0_to_9,
  r#"
    for i in 0..10:
      print i
  "#
}
check! {
  for_print_0_to_10,
  r#"
    for i in 0..=10:
      print i
  "#
}
check! {
  for_print_nothing,
  r#"
    for i in 10..0:
      print i
  "#
}
check! {
  for_range_vars,
  r#"
    start := 0
    end := 10
    for i in start..end:
      print i
  "#
}
check! {
  for_range_vars_inclusive,
  r#"
    start := 0
    end := 10
    for i in start..=end:
      print i
  "#
}
check! {
  for_print_odd_0_to_9,
  r#"
    for i in 0..10:
      if i % 2 == 0: continue
      print i
  "#
}
