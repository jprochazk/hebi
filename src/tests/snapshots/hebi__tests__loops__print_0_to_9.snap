---
source: src/tests/loops.rs
expression: snapshot
---
# Modules:


# Input:
i := 0
loop:
  if i >= 10:
    break
  print i
  i += 1


# Result (success):
true

# Stdout:
0
1
2
3
4
5
6
7
8
9


